core/arch/arm/mm/tee_mm.c

```diff
@@ -115,15 +117,22 @@ tee_mm_entry_t *tee_mm_alloc(tee_mm_pool_t *pool, uint32_t size)
 
 	/* check if we have enough memory */
 	if (pool->flags & TEE_MM_POOL_HI_ALLOC) {
-		if (((entry->offset << pool->shift) - pool->lo) < size)
+		/*
```

Comment:
I think we are sure there is enough memory when `entry->next != NULL`. So we could test this before the test `if (pool->flags & TEE_MM_POOL_HI_ALLOC)`